api/valor_api/backend/metrics/metric_utils.py

```diff
@@ -553,10 +569,46 @@ def prepare_filter_for_evaluation(
         )
     )
 
+    datum_filter = filters.model_copy()
+    datum_filter.groundtruths = None
+    datum_filter.predictions = None
```

Comment:
This datum filter is concerning to me as it could include datums outside what the user specifies.

To perform datum queries i would use the groundtruth_filter with `label_source=models.Annotation`

Reply:
@czaloom What if the datums don't have any labels?

Reply:
It wont remove them, it just sets the source if labels are specified.